docs/_tooling/sphinx_extensions/sphinx_extensions/build/modularity/modularity.py

```diff
@@ -0,0 +1,87 @@
+from pprint import pprint
+from sphinx.application import Sphinx
+from sphinx_needs.data import SphinxNeedsData, NeedsInfoType
+from sphinx_needs.config import NeedsSphinxConfig
+from sphinx.environment import BuildEnvironment
+from sphinx.util import logging
+from copy import deepcopy
+import os
+import json
+
+logger = logging.getLogger(__name__)
+
+
+def read_filter_tags(app: Sphinx):
+    """
+    Helper function to read in the 'filter_tags' provided by `feature_flag.bzl`.
+
+    Args:
+        app: The current running Sphinx-Build application
+
+    Returns:
+       - List of tags e.g. ['some-ip','another-tag']
+       OR
+       - Empty list if there is an exception when reading the file.
+
+    Errors:
+        If 'filter_tags_file_path' can't be found in the config
+    """
+    # asserting our worldview
+    assert hasattr(
+        app.config, "filter_tags_file_path"
+    ), "Config missing filter_tags_file_path, this is mandatory."
+    print(f"Attempting to read filter tags from: {app.config.filter_tags_file_path}")
```

Comment:
Are print statements actually readable when running sphinx?
```suggestion
    logger.debug(f"Attempting to read filter tags from: {app.config.filter_tags_file_path}")
```

Reply:
Added several log statements in newest commit